### DISCUSSION TITLE
Map projection question

### DISCUSSION THREAD

**Discussion by bathoorn** [skyfielders/python-skyfield#917] [Q&A]
Hi,

For fun i was trying to draw constellation borders on a starmap i created earlier.

Drawing and projecting stars is easier as they are just dots. So the projection is easier.
When i want to draw constellation borders i have edges of the lines in ra/dec but when i project these points
the line will be straight while in reality they should be curved lines after the projection.

As an example my starmap so far
![summer-triangle](https://github.com/skyfielders/python-skyfield/assets/760615/db4cdcbf-9675-4df1-9862-8dfc6ce27fe6)

You can see that some of the borders do not touch as the lines are not drawn correctly.
So this happens because i do a projection of the edges of the constellation border.

`
from skyfield.api import Star, load, wgs84, N, S, W, E
from skyfield.projections import build_stereographic_projection
from datetime import datetime
from pytz import timezone

AMS = timezone('Europe/Amsterdam')
ts = load.timescale()
t = ts.from_datetime(AMS.localize(datetime(2023, 11, 15, 22, 0, 0)))
degrees = 0.0
amsterdam = wgs84.latlon(lat, lon, elevation_m=28).at(t)
position = amsterdam.from_altaz(alt_degrees=90, az_degrees=degrees)
projection = build_stereographic_projection(position)
cons_borders = earth.at(t).observe(Star.from_dataframe(constdata))
constdata['x'], constdata['y'] = projection(cons_borders)
`

Where i should be doing a projection of the line itself so that all points on the line are projected. But my understanding of the projection function is not good enough to know if that is possible.

Can anybody here help?
As i could do interpolation on the lines and divide it in smaller segments, but maybe there is a better solution.

Thanks,
Ronnie

**Comment by bathoorn** [comment]
interpolation fixes it a little
but it messes with the dashed lines :)

![summer-triangle](https://github.com/skyfielders/python-skyfield/assets/760615/47c6072c-d4fa-4cec-a649-4bf143ce2acb)

**Comment by bathoorn** [comment]
It looks like I might be ble to use this
https://scitools.org.uk/cartopy/docs/latest/
It is able to do projections of points and lines.

It is created for earth maps. But it looks like it can als be used for astronomy maps.

**Comment by bathoorn** [comment]
Got it to look better
- cut off at the horizon
- blue background
- planets and the moon visible

![summer-triangle](https://github.com/skyfielders/python-skyfield/assets/760615/508cd357-33ed-4c27-a657-4c9a64567616)

You can find the code here
https://github.com/bathoorn/starmap/blob/main/starmap.py

**Reply by brandon-rhodes** [reply]
I'm glad you got it working better! Did you wind up doing interpolation to draw the lines, or was it cartopy that produced this latest image? To answer your earlier question, I think interpolation is the way all libraries would solve this problem — at least down at the implementation level, even if they provide a simpler interface that doesn't make the caller perform the interpolation — and I suspect that there's a way, in whichever library you're using, to build a line-made-of-smaller-line-segments that gets dashes drawn along it where the dashes don't start and stop again each time it reaches an interpolation point.

**Reply by bathoorn** [reply]
I indeed dit interpolation on the last image. And at some angles that makes the dashes disappear. It looks like maybe matplotlib itself can also do different projections. Maybe with polar projection I can already do a lot with a simple star map.
